packages/permission-controller/src/rpc-methods/revokePermissions.test.ts

```diff
@@ -0,0 +1,158 @@
+      implementation(req as any, res as any, next, end, {
+        revokePermissionsForOrigin: mockRevokePermissionsForOrigin,
+      }),
+    );
+
+    const req = {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'wallet_revokePermissions',
+      params: {},
+    };
+
+    const expectedError = rpcErrors
+      .invalidParams({
+        data: { request: { ...req } },
+      })
+      .serialize();
+    delete expectedError.stack;
+
+    const response: any = await engine.handle(req as any);
+    delete response.error.stack;
+    expect(response.error).toStrictEqual(expectedError);
+    expect(mockRevokePermissionsForOrigin).not.toHaveBeenCalled();
+  });
+
+  it('returns an error if the permissionKeys is a plain object', async () => {
+    const { implementation } = revokePermissionsHandler;
+    const mockRevokePermissionsForOrigin = jest.fn();
+
+    const engine = new JsonRpcEngine();
+    engine.push((req, res, next, end) =>
+      implementation(req as any, res as any, next, end, {
+        revokePermissionsForOrigin: mockRevokePermissionsForOrigin,
+      }),
+    );
+
+    const req = {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'wallet_revokePermissions',
+      params: [{}],
+    };
+
+    const expectedError = rpcErrors
+      .invalidParams({
+        data: { request: { ...req } },
+      })
+      .serialize();
+    delete expectedError.stack;
+
+    const response: any = await engine.handle(req as any);
+    delete response.error.stack;
+    expect(response.error).toStrictEqual(expectedError);
+    expect(mockRevokePermissionsForOrigin).not.toHaveBeenCalled();
+  });
+
+  it('returns an error if the params are not set', async () => {
+    const { implementation } = revokePermissionsHandler;
+    const mockRevokePermissionsForOrigin = jest.fn();
+
+    const engine = new JsonRpcEngine();
+    engine.push((req, res, next, end) =>
+      implementation(req as any, res as any, next, end, {
+        revokePermissionsForOrigin: mockRevokePermissionsForOrigin,
+      }),
+    );
+
+    const req = {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'wallet_revokePermissions',
+    };
+
+    const expectedError = rpcErrors
+      .invalidParams({
+        data: { request: { ...req } },
+      })
+      .serialize();
+    delete expectedError.stack;
+
+    const response: any = await engine.handle(req as any);
+    delete response.error.stack;
+    expect(response.error).toStrictEqual(expectedError);
+    expect(mockRevokePermissionsForOrigin).not.toHaveBeenCalled();
+  });
+
+  it('returns an error if the request params is an empty array', async () => {
+    const { implementation } = revokePermissionsHandler;
+    const mockRevokePermissionsForOrigin = jest.fn();
+
+    const engine = new JsonRpcEngine();
+    engine.push((req, res, next, end) =>
+      implementation(req as any, res as any, next, end, {
+        revokePermissionsForOrigin: mockRevokePermissionsForOrigin,
+      }),
+    );
```

Comment:
can the test setup be moved into a `beforeEach`?

Reply:
I think we've been preferring a setup function. I think we should do this but it shouldn't be blocking merging this PR